presto-orc/src/main/java/com/facebook/presto/orc/reader/SelectiveStreamReaders.java

```diff
@@ -61,7 +61,10 @@ public static SelectiveStreamReader createStreamReader(
                 return new LongSelectiveStreamReader(streamDescriptor, getOptionalOnlyFilter(type, filters), outputType, systemMemoryContext);
             }
             case FLOAT:
+                throw new IllegalArgumentException("Unsupported type: " + streamDescriptor.getStreamType());
             case DOUBLE:
+                checkArgument(requiredSubfields.isEmpty(), "Double stream reader doesn't support subfields");
```

Comment:
Also -- on line 114 / 111, shouldn't there be a change there too? I just noticed it.